### ISSUE TITLE
Model is validated only after change

### ISSUE THREAD

**Issue by dimrsilva** [angular-ui/ui-mask#1]
I need to set the mask depending on a previous field.
So, I created a scope var to the mask definition, but the field is only validated after first interaction.

Here is a plunk: http://plnkr.co/edit/7Sk9N1x8LjJXakAtRLxb?p=preview

Apparently, the model value needs to be updated for the validation works.
I did a little trick for that on my client code:

* change model value
* trigger `$scope.$apply()`
* restore model value
* trigger `$scope.$apply()`

```javascript
var original = $scope.modelValue;
$scope.modelValue = '';
$scope.$apply();
$scope.modelValue = doc;
$scope.$apply();
```

Original issue on ui-utils: https://github.com/angular-ui/ui-utils/issues/351

**Comment by PowerKiKi** [comment]
Thanks for taking the time to test and re-submit your issue. Since we don't have much resources, and I am not familiar with this particular code base, it would be much faster if you were able to submit a Pull Request. Otherwise you may have to wait a bit longer ...

**Comment by dimrsilva** [comment]
Ok, I'll try to work on it.

**Comment by PowerKiKi** [comment]
Closed via #3
